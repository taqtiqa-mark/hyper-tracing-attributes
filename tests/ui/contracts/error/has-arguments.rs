use http_tracing_attributes::contracts;

#[contracts(a, b)]
fn f() {}

fn main() {}
