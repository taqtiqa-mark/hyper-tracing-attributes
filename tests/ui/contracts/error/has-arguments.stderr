warning: unused import: `crate::utility::to_token_stream`
 --> src/lib.rs
  |
  | use crate::utility::to_token_stream;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: variable does not need to be mutable
  --> src/server_send/analyze.rs
   |
   |     let mut level: syn::ExprPath = syn::parse_quote!(Level::DEBUG);
   |         ----^^^^^
   |         |
   |         help: remove this `mut`
   |
   = note: `#[warn(unused_mut)]` on by default

warning: variable does not need to be mutable
  --> src/server_send/analyze.rs
   |
   |     let mut skip: syn::ExprArray = syn::parse_quote!([]);
   |         ----^^^^
   |         |
   |         help: remove this `mut`

warning: variable does not need to be mutable
  --> src/lib.rs
   |
   |     let mut meta_args = syn::parse_macro_input!(meta as crate::server_send::parse::Args);
   |         ----^^^^^^^^^
   |         |
   |         help: remove this `mut`

warning: variable does not need to be mutable
  --> src/lib.rs
   |
   |     let mut meta_args2: syn::DeriveInput = syn::parse2(meta2.into()).expect("Ast2");
   |         ----^^^^^^^^^^
   |         |
   |         help: remove this `mut`

warning: function is never used: `get_skip`
   --> src/server_send/analyze.rs
    |
    | fn get_skip(skip: &mut syn::ExprArray, span: proc_macro2::Span, tokens: proc_macro2::TokenStream) {
    |    ^^^^^^^^
    |
    = note: `#[warn(dead_code)]` on by default

warning: function is never used: `to_token_stream`
 --> src/utility/mod.rs
  |
  | pub(crate) fn to_token_stream(
  |               ^^^^^^^^^^^^^^^

error: this attribute takes no arguments

         = help: use `#[contracts]`

 --> tests/ui/contracts/error/has-arguments.rs:3:1
  |
3 | #[contracts(a, b)]
  | ^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `contracts` (in Nightly builds, run with -Z macro-backtrace for more info)
